### PR TITLE
Remove the cluster max size check from openstorage

### PR DESCRIPTION
**Pull request by adityadani**
**What this PR does / why we need it**:
- Let the Storage Listeners decide the maximum no. of nodes that can be
  added to the cluster.
- Add a new StorageListener API CanNodeJoin that gets invoked before joining a node
  to a cluster.

Signed-off-by: Aditya Dani <aditya@portworx.com>

### CODE REVIEW

**Comment by zoxpx** [comment]
@adityadani -- the travis-build is failing, looks like `cluster/mock/cluster.mock.go` will need update

**Comment by adityadani** [comment]
@zoxpx I have updated the PR

File: cluster/manager/manager.go [hunk 1]
```diff
@@ -1135,25 +1112,21 @@ func (c *ClusterManager) initListeners(
...
-		if clusterMaxSize > 0 && numNodes > clusterMaxSize {
-			// throw a slightly different error depending if node was already
-			// in the cluster.
-			if exist {
-				return ErrOldNodeOverMaxCapacity(clusterMaxSize)
+	initFunc := func(clusterInfo *cluster.ClusterInfo) error {
+		// Irrespective of whether the node is doing an Init or is
+		// already in cluster, check with listeners if it is OK to join
+		// this cluster.
+		for e := c.listeners.Front(); e != nil; e = e.Next() {
```
Comment (sangleganesh) [inline cluster/manager/manager.go]: there is already a finalizeCB callback, so not sure why we want to callback listeners again ?

Comment (adityadani) [reply cluster/manager/manager.go]: We want the clusterMaxSize check to be called during Init as well as Join. This piece of code (initFunc) gets called during both. While in Init finalizeCbs is populated, but in Join it is an empty array.

I have added a comment below about finalizeCbs being nil

**Pull request merged**